e2e/helpers/http.ts

```diff
@@ -1,141 +1,107 @@
-import { createHmac } from 'crypto'
-import { fetchWithExponentialTimeout } from './srcFetch'
+import { Page, expect } from '@playwright/test'
+import { navigateToHomepageIfPageHasNotBeenLoaded } from './localStorage'
 
-export const BASE64_ENCODING = 'base64'
-export const DEFAULT_GETDELETE_HEADERS = {
-  Accept: '*/*',
-  'User-Agent': 'SynapseWebClient',
-}
-export const DEFAULT_POST_HEADERS = {
-  ...DEFAULT_GETDELETE_HEADERS,
-  Accept: 'application/json; charset=UTF-8',
-  'Content-Type': 'application/json; charset=UTF-8',
-}
-
-function generateDigitalSignature(
-  data: string,
-  base64EncodedSecretKey: string,
-) {
-  const hash = createHmac(
-    'sha1',
-    Buffer.from(base64EncodedSecretKey, BASE64_ENCODING),
-  )
-    .update(data)
-    .digest(BASE64_ENCODING)
-  return hash
-}
-
-function getDigitalSignature(
-  uri: string,
-  username: string,
-  base64EncodedSecretKey: string,
-) {
-  const timestamp = new Date().toISOString()
-  const signature = generateDigitalSignature(
-    `${username}${uri}${timestamp}`,
-    base64EncodedSecretKey,
-  )
-  return {
-    userId: username,
-    signatureTimestamp: timestamp,
-    signature: signature,
-  }
-}
-
-export function getUserIdFromJwt(token: string) {
-  const payload = JSON.parse(
-    Buffer.from(token.split('.')[1], BASE64_ENCODING).toString(),
-  )
-  return payload.sub
+export enum BackendDestinationEnum {
+  REPO_ENDPOINT,
+  PORTAL_ENDPOINT,
 }
 
-function updateHeaders(
-  headers: { [key: string]: string },
-  uri: string,
-  accessToken?: string,
-  userName?: string,
-  apiKey?: string,
-) {
-  return {
-    ...headers,
-    ...(accessToken && {
-      'Access-Control-Request-Headers': 'authorization',
-      Authorization: `Bearer ${accessToken}`,
-    }),
-    ...(apiKey && userName && getDigitalSignature(uri, userName, apiKey)),
-  }
+export async function waitForSrcEndpointConfig(page: Page) {
+  // window only available after page has initially loaded
+  await navigateToHomepageIfPageHasNotBeenLoaded(page)
+  // ensure that endpoint config is set,
+  // ...so API calls point to the correct stack
+  await expect(async () => {
+    const response = await page.evaluate('window.SRC.OVERRIDE_ENDPOINT_CONFIG')
+    expect(response).not.toBeUndefined()
+  }).toPass()
 }
 
 export async function doPost<T>(
-  endpoint: string,
-  uri: string,
-  requestContent: string,
-  accessToken?: string,
-  userName?: string,
-  apiKey?: string,
+  page: Page,
```

Comment:
Instead of passing a `Page` around, you could consider configuring the test process to point to the correct stack using an environment variable, probably by reading the Maven settings.xml so there's just one source of truth. Then I think a lot of this code could run without requiring the browser and loaded page.

You wouldn't have confidence that the `SRC.OVERRIDE_ENDPOINT_CONFIG` is properly set, but I don't think that's such a big deal.

What do you think?

Reply:
I like that idea for looking up the correct stack! Since we'll still need to evaluate the page to use SRC's HttpClient and SynapseClient, I might hold off on implementing that for now. However, I've opened [a ticket](https://sagebionetworks.jira.com/browse/SWC-6550) to track the work and capture this idea!